CoreUtilities/src/au/gov/asd/tac/constellation/utilities/datastructure/FourTuple.java

```diff
@@ -0,0 +1,198 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package au.gov.asd.tac.constellation.utilities.datastructure;
+
+import java.io.Serializable;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+/**
+ * A data-structure for storing a triplet of related values.
+ *
+ * @author arcturus
+ * @author cygnus_x-1
+ *
+ * @param <A> the type of the first object in the 4-tuple
+ * @param <B> the type of the second object in the 4-tuple
+ * @param <C> the type of the third object in the 4-tuple
+ * @param <D> the type of the fourth object in the 4-tuple
+ */
+public class FourTuple<A, B, C, D> implements Serializable, Comparable<FourTuple<A, B, C, D>> {
+
+    private static final long serialVersionUID = 1L;
+
+    private A first;
+    private B second;
+    private C third;
+    private D fourth;
+
+    public FourTuple(final A first, final B second, final C third, final D fourth) {
+        this.first = first;
+        this.second = second;
+        this.third = third;
+        this.fourth = fourth;
+    }
+
+    /**
+     * Convenience method for creating an appropriately typed tuple.
+     *
+     * @param <A> the type of the first object in the 4-tuple
+     * @param <B> the type of the second object in the 4-tuple
+     * @param <C> the type of the third object in the 4-tuple
+     * @param <D> the type of the third object in the 4-tuple
+     * @param first the first object in the tuple
+     * @param second the second object in the tuple
+     * @param third the third object in the tuple
+     * @param fourth the fourth object in the tuple
+     * @return a 4-tuple that is templated with the types of first, second,
+     * third and fourth
+     */
+    public static <A, B, C, D> FourTuple<A, B, C, D> create(A first, B second, C third, D fourth) {
+        return new FourTuple<>(first, second, third, fourth);
+    }
+
+    public A getFirst() {
+        return first;
+    }
+
+    public void setFirst(final A first) {
+        this.first = first;
+    }
+
+    public B getSecond() {
+        return second;
+    }
+
+    public void setSecond(final B second) {
+        this.second = second;
+    }
+
+    public C getThird() {
+        return third;
+    }
+
+    public void setThird(final C third) {
+        this.third = third;
+    }
+
+    public D getFourth() {
+        return fourth;
+    }
+
+    public void setFourth(final D fourth) {
+        this.fourth = fourth;
+    }
+
+    public Stream<Object> stream() {
+        return Stream.of(first, second, third, fourth);
+    }
+
+    @Override
+    public boolean equals(final Object obj) {
+        if (!(obj instanceof FourTuple)) {
```

Comment:
was there a reason for formulating the if conditional this way as appose to `this.getClass() != obj.getClass()` (which is what I see more commonly in equals functions)

Reply:
Good pickup and after reading https://stackoverflow.com/a/4989843 I have gone with using `getClass`. Updated for all the Tuple classes.